heidrun/nara_json.rb

```diff
@@ -0,0 +1,575 @@
+def nara_catalog_uri(id)
+  "http://catalog.archives.gov/id/#{id.node}"
+end
+
+# See object and preview mappings
+def make_obj_uri(obj)
+  URI::escape(obj.node['@url'])  # Can have space characters
+end
+def make_obj_dcformat(obj)
+  obj.node['@mime']
+end
+def make_preview_uri(obj)
+  URI::escape(obj.node['@url'])  # Can also have space characters
+end
+def make_preview_dcformat(obj)
+  obj.node['@mime']
+end
+
+# Return a string suitable for sourceResource.contributor or
+# sourceResource.publisher.
+#
+# @see #make_contributor
+# @see #make_publisher
+# @param contributor_array [Krikri::JsonParser::Value]
+# @return [String, RDF::Literal]
+def contributor_term_name(contributor_array)
+  node = contributor_array.node
+  contributors = node.fetch('organizationalContributor',
+                            node['personalContributor'])
+  contributors = [contributors] unless contributors.is_a? Array
+
+  yield(contributors)
+
+  return contributors.first['contributor']['termName'] \
+    unless contributors.empty?
+
+  # FIXME:  It's not clear how to indicate the concept of "nothing" to
```

Comment:
By putting the filtering code inside of these methods I was trying to keep the statements for each field shorter, so I understand how `select` or `reject` could probably be used lower down, but my concern then is that those statements would become more complex.  Also, the `contributor` field requires that if there are more than one Publisher, that the "Most Recent" one be used, so that seemed like a logic more complex than I'd want to put in one of those chained expressions.